background_mapping_handler.js

```diff
@@ -48,13 +53,13 @@ browser.runtime.onInstalled.addListener(() => {
  * Handle all messages received by this background script, called whenever a
  * message is sent by any other endpoint using browser.runtime.sendMessage()
  * @param {*} message - Data object sent by message sender
- * @param {*} sender // not used
- * @param {*} sendResponse // not used
  * @returns {Promise<Object>} A promise representing an Object sent as a response to the message
  */
-function messageListener(message, sender, send_response) {
+function messageListener(message) {
 	if (message.intended_recipient !== RECIPIENT_BACKGROUND) {
-		return; // returning nothing does not send a response, allowing other recipients of this message to still respond
+		// returning nothing does not send a response,
+		// allowing other recipients of this message to still respond
+		return;

```

Comment:
If Smeagol expects a promise to be returned, what does returning nothing do? Would it not cause an error when a `.then` is called on the return value of messageListener? 

Does this not come up today because all messages are sent to the background script?

Reply:
Yes and yes :(